src/Analyzer/ValidationUtils.cpp

```diff
@@ -134,15 +134,70 @@ class ValidateGroupByColumnsVisitor : public ConstInDepthQueryTreeVisitor<Valida
     }
 
 private:
+
+    static bool areColumnSourcesEqual(const QueryTreeNodePtr & lhs, const QueryTreeNodePtr & rhs)
```

Comment:
Let's remove this function, instead I propose to add a new flag `compare_column_sources` into `CompareOptions` and use it in `ColumnNode::isEqualImpl`.